controllers/networkfence_controller.go

```diff
@@ -58,5 +155,78 @@ func (r *NetworkFenceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 func (r *NetworkFenceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&csiaddonsv1alpha1.NetworkFence{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// validateNetworkFenceSpec validates the NetworkFence spec and checks if values are neither nil nor empty.
+func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
+	if nwFence.Spec.Driver == "" {
+		return errors.New("required parameter 'driver' is not specified")
+	}
+	if nwFence.Spec.Secret.Name == "" {
+		return errors.New("required parameter `secret.name` is not specified")
+	}
+	if nwFence.Spec.Secret.Namespace == "" {
+		return errors.New("required parameter `secret.namespace` is not specified")
+	}
+	if nwFence.Spec.Cidrs == nil {
+		return errors.New("required parameter 'cidrs' is not specified")
+	}
+	return nil
+}
+
+// sendUnfenceNetworkReq sends a UnfenceClusterNetwork request.
+func (r *NetworkFenceReconciler) sendUnfenceNetworkReq(ctx context.Context, nwFence *csiaddonsv1alpha1.NetworkFence, controllerClient *grpc.ClientConn, logger logr.Logger) error {
```

Comment:
do same for sendFenceRequest also? and this should be a function not a method.

Reply:
Updated now